### PR TITLE
New Basemaps: Search and Symbology Categories

### PR DESCRIPTION
**Pull request by jenmerritt**
### Description

This PR is for the following sample categories:
- Search
- Symbology

Notes specific to this PR: 
- Find Address, Find Place, Reverse Geocode Online - also had geocode service URLs updated.
Note: It's been previously raised in a Review by Rachael that the About the Data section in Find Place could be in the other 2 samples too - this was added to the designs in Common Samples with some other tweaks, but for now I've just updated the existing URL so that it is 'correct' as there will need to be a little more discussion on this again in the Samples team since this has changed again since then.

### General comments on Basemap updates:

It updates `Map` and/or `Basemap` Constructors as per the new design. This may include the following changes depending on the sample:
- Replace `Basemap.Type` with `BasemapStyle`
- If a lat/long/zoom was being set in the `Map` constructor, this is replaced with a `Viewpoint` being set on the `MapView`.
- Any Readme Updates identified

In addition, the samples that include the above changes, or other services requiring an API Key going forward, have the API Key  set within the sample. This will then work in conjunction with the Gradle Task being added in PR 593.

Once 593 is approved and merged into `dev` we will merge those changes into these PRs and then update all the Samples to `100.10.0-xxxx` and do a round of testing of the Samples with all the API Key / Basemap changes at that time. (We have left them as 100.9.0 (in most cases) for now to minimize merge conflicts in the `build.gradle` file).

In the meantime, the updates listed above can be reviewed. Thanks!

### CODE REVIEW

File: search/find-place/src/main/java/com/esri/samples/find_place/FindPlaceController.java [hunk 1]
```diff
@@ -67,9 +68,14 @@
 
   @FXML
   public void initialize() {
+    // authentication with an API key or named user is required to access basemaps and other location services
+    String yourAPIKey = System.getProperty("apiKey");
+    ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
 
-    // create ArcGISMap with streets basemap and add it to map view
-    ArcGISMap map = new ArcGISMap(Basemap.createStreets());
+    // create ArcGISMap with streets basemap style
```
Comment (JonLavi) [inline search/find-place/src/main/java/com/esri/samples/find_place/FindPlaceController.java]: We could plain-english this?
```suggestion
    // create a map with the streets basemap style
```

File: search/find-place/src/main/java/com/esri/samples/find_place/FindPlaceController.java [hunk 2]
```diff
@@ -82,7 +88,7 @@ public void initialize() {
     callout.setLeaderPosition(Callout.LeaderPosition.BOTTOM);
 
     // create a locatorTask task
```
Comment (JonLavi) [inline search/find-place/src/main/java/com/esri/samples/find_place/FindPlaceController.java]: Just noticing this comment as well:
```suggestion
    // create a locator task
```

File: symbology/picture-marker-symbol/src/main/java/com/esri/samples/picture_marker_symbol/PictureMarkerSymbolSample.java [hunk 3]
```diff
@@ -69,15 +70,19 @@ public void start(Stage stage) {
...
       stage.show();
 
-      // create a ArcGISMap with the topographic basemap
-      map = new ArcGISMap(Basemap.createTopographic());
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
 
-      // create view for this map
+      // create a ArcGISMap with the topographic basemap style
```
Comment (JonLavi) [inline symbology/picture-marker-symbol/src/main/java/com/esri/samples/picture_marker_symbol/PictureMarkerSymbolSample.java]:
```suggestion
      // create a map with the topographic basemap style
```

File: symbology/simple-fill-symbol/src/main/java/com/esri/samples/simple_fill_symbol/SimpleFillSymbolSample.java [hunk 4]
```diff
@@ -80,17 +85,18 @@ public void start(Stage stage) {
...
+      final ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC);
 
-      // create a view for this ArcGISMap and set ArcGISMap to it
+      // create a map view and set its map
       mapView = new MapView();
       mapView.setMap(map);
 
+      // set a viewpoint on the map view
+      Point initialPoint = new Point(-12000000, 5400000, SpatialReferences.getWebMercator());
+      Viewpoint viewpoint = new Viewpoint(initialPoint, 10000000); // point, scale
```
Comment (JonLavi) [inline symbology/simple-fill-symbol/src/main/java/com/esri/samples/simple_fill_symbol/SimpleFillSymbolSample.java]: We could probably get rid of this `initialPoint` variable and just put it straight in the `Viewpoint` constructor.
Either way, we could definitely get rid of the comment at the back of the line creating the `Viewpoint` which describes the input parameters.
```suggestion
      Viewpoint viewpoint = new Viewpoint(new Point(-12000000, 5400000, SpatialReferences.getWebMercator()), 10000000);
```

File: symbology/simple-line-symbol/src/main/java/com/esri/samples/simple_line_symbol/SimpleLineSymbolSample.java [hunk 5]
```diff
@@ -74,17 +79,18 @@ public void start(Stage stage) {
...
+      final ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_IMAGERY_STANDARD);
 
-      // create a view for this ArcGISMap and set ArcGISMap to it
+      // create a map view and set its map
       mapView = new MapView();
       mapView.setMap(map);
 
+      // set a viewpoint on the map view
+      Point point = new Point(-226773, 6550477, SpatialReferences.getWebMercator());
+      Viewpoint viewpoint = new Viewpoint(point, 7200); // point, scale
```
Comment (JonLavi) [inline symbology/simple-line-symbol/src/main/java/com/esri/samples/simple_line_symbol/SimpleLineSymbolSample.java]: Same suggestion as above, at the very least let's ditch the comment.

File: symbology/simple-marker-symbol/src/main/java/com/esri/samples/simple_marker_symbol/SimpleMarkerSymbolSample.java [hunk 6]
```diff
@@ -51,8 +52,12 @@ public void start(Stage stage) {
...
-      // create ArcGISMap with imagery basemap
-      final ArcGISMap map = new ArcGISMap(Basemap.createImagery());
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create ArcGISMap with imagery basemap style
+      final ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_IMAGERY_STANDARD);
 
       // create spatial reference for WGS 1948
```
Comment (JonLavi) [inline symbology/simple-marker-symbol/src/main/java/com/esri/samples/simple_marker_symbol/SimpleMarkerSymbolSample.java]: If you're feeling up for it, you could clean up lines 62-67 as well to match the previous samples in this review?

File: symbology/simple-renderer/src/main/java/com/esri/samples/simple_renderer/SimpleRendererSample.java [hunk 7]
```diff
@@ -52,8 +53,12 @@ public void start(Stage stage) {
       stage.setScene(scene);
       stage.show();
 
-      // create a ArcGISMap with the imagery basemap
-      final ArcGISMap map = new ArcGISMap(Basemap.createImageryWithLabels());
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a ArcGISMap with the imagery basemap style
```
Comment (JonLavi) [inline symbology/simple-renderer/src/main/java/com/esri/samples/simple_renderer/SimpleRendererSample.java]:
```suggestion
      // create a map with the imagery basemap style
```

File: symbology/unique-value-renderer/src/main/java/com/esri/samples/unique_value_renderer/UniqueValueRendererSample.java [hunk 8]
```diff
@@ -62,8 +63,12 @@ public void start(Stage stage) {
       stage.setScene(scene);
       stage.show();
 
-      // create ArcGISMap with topographic basemap
-      final ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create ArcGISMap with topographic basemap style
```
Comment (JonLavi) [inline symbology/unique-value-renderer/src/main/java/com/esri/samples/unique_value_renderer/UniqueValueRendererSample.java]:
```suggestion
      // create map with the topographic basemap style
```

**Review by JonLavi — APPROVED** [review]
LGTM @jenmerritt , just a couple of similar sugestions on plain-englishness and consistency

File: symbology/simple-fill-symbol/src/main/java/com/esri/samples/simple_fill_symbol/SimpleFillSymbolSample.java (hunk 4 above)
Comment (jenmerritt) [reply symbology/simple-fill-symbol/src/main/java/com/esri/samples/simple_fill_symbol/SimpleFillSymbolSample.java]: Have removed the `Point` and just set the `Viewpoint`.

File: symbology/simple-line-symbol/src/main/java/com/esri/samples/simple_line_symbol/SimpleLineSymbolSample.java (hunk 5 above)
Comment (jenmerritt) [reply symbology/simple-line-symbol/src/main/java/com/esri/samples/simple_line_symbol/SimpleLineSymbolSample.java]: Removed `Point` and just set `Viewpoint`

File: symbology/simple-marker-symbol/src/main/java/com/esri/samples/simple_marker_symbol/SimpleMarkerSymbolSample.java (hunk 6 above)
Comment (jenmerritt) [reply symbology/simple-marker-symbol/src/main/java/com/esri/samples/simple_marker_symbol/SimpleMarkerSymbolSample.java]: Tidied up

File: symbology/unique-value-renderer/src/main/java/com/esri/samples/unique_value_renderer/UniqueValueRendererSample.java (hunk 8 above)
Comment (jenmerritt) [reply symbology/unique-value-renderer/src/main/java/com/esri/samples/unique_value_renderer/UniqueValueRendererSample.java]: Made a few changes to this one on the `Viewpoint` side of things too

**Review by Rachael-E — APPROVED** [review]
@jenmerritt no further comments here, looks good!

**Comment by jenmerritt** [comment]
`dev` merged in (including API Key gradle tasks changes). Samples that have an API Key set have been tested running from the IDE.

Await merging these changes until release.

**Pull request merged**